src/wallet/spend.cpp

```diff
@@ -1349,6 +1353,9 @@ bool FundTransaction(CWallet& wallet, CMutableTransaction& tx, CAmount& nFeeRet,
     // Set the user desired locktime
     coinControl.m_locktime = tx.nLockTime;
 
+    // Set the user desired version
+    coinControl.m_version = tx.nVersion;
```

Comment:
nit:
it seems that the version field is not sanitized. The user provided `tx.nVersion` is a int32_t while `coinControl.m_version` is a uint32_t.